### PR TITLE
Added support for Ubuntu 18 04

### PR DESCRIPTION
**Pull request by adrianherrera**

### CODE REVIEW

File: determine_clang_binary_suffix.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#
+# Inspects the current operating system and determines which Clang binary to
+# download. If a valid version is found, the suffix of the file to download is
+# printed to stdout. Otherwise an error message is printed to stderr.
+#
+# Note: This script is only really meant to be used by the S2E Makefile. It has
+# no real use outside of this.
+#
+# Copyright (C) 2018 Adrian Herrera
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
```
Comment (vitalych) [inline determine_clang_binary_suffix.py]: Could you add the license header to the beginning and swich to MIT?

File: determine_clang_binary_suffix.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+def _get_ubuntu_version(version_string):
+    """
+    Determine the Clang binary to downoad from the version string returned by
+    ``platform.linux_distribution``.
+    """
+    major_version, minor_version = map(int, version_string.split('.'))
+
+    # Currently S2E only supports LLVM 3.9.1, and the only Clang binary
+    # packages that exist for this version are for Ubuntu 14.04 and 16.04
+    if major_version == 14 and minor_version >= 4:
```
Comment (vitalych) [inline determine_clang_binary_suffix.py:70]: I suppose we don't need to check this if we drop support for 14.

**Comment by vitalych** [comment]
Did you check that the demo docker image builds and runs? I promise I'll work on Jenkins when I return from vacation :)

**Review by vitalych — CHANGES_REQUESTED** [review]
Looks great, few comments to address.

**Comment by adrianherrera** [comment]
Actually no, the Docker containers don't work. Having the additional Python script throws a spanner in the works. I don't really want to pollute the `$(S2E_DIR)/source/s2e` directory with more symlinks (in fact I would argue that `Makefile.docker`, `Dockerfile`, `Dockerfile.dist`, etc. shouldn't be there anyway), so I'm trying to refactor a bit.

Hopefully I'll have a elegant solution soon. Maybe even before you finish your vacation :)

**Comment by adrianherrera** [comment]
Okay, Docker demo is working fine.

All PRs are good to go 🚀

**Review by vitalych — APPROVED** [review]

**Pull request merged**